### PR TITLE
chore: setup dependabot

### PR DESCRIPTION
**Pull request by P0lip**
Feel free to adjust it to your liking.

**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# https://docs.github.com/en/github/administering-a-repository/configuration-options-for-dependency-updates
+version: 2
+
+updates:
+  - package-ecosystem: "npm"
```
Comment (P0lip) [inline .github/dependabot.yml:5]: Unfortunately, sharing multiple confiruations for the same package ecosystem is not supported yet :/
There have been multiple requests for it, but AFAIK it's not implemented just yet.
See https://github.com/stoplightio/spectral/pull/1247#issue-438571463

Comment (nulltoken) [reply .github/dependabot.yml:5]: @P0lip I've pushed an update that should make dependabot happy and able to detect outdated deps not only at the first level, but also within the `client` and `server` folders.

**TL;DR**
It's a gigantic hack. I'll need your help to get rid of webpack and spread some of your rollup magic onto this

**Gory details**
- We use yarn workspaces, so that dependabot can find all the package.json files, provided the doco doesn't lie (cf. https://dependabot.com/blog/simpler-javascript-support/)
- We leverage "nohoist" (actually killing most of the workspaces benefits) so that webpack is happy and can help us reduce the size of the final vsix package by finding the modules the server don't really need
- Most of the work in that change is to bridge the gap between `vsce` expectations about what a proper VSCode extension tree structure should look like and the impact of switching to workspaces

Comment (nulltoken) [reply .github/dependabot.yml:5]: @P0lip Although you may not be familiar with the existing codebase, I'd really appreciate if you could peek at the proposed changes.

Comment (nulltoken) [reply .github/dependabot.yml:5]: @P0lip Fixed

File: make.js [hunk 2]
```diff
@@ -20,9 +20,13 @@ const outputPath = {
...
   banner('Target: AllDev');
 
-  const backupPath = await patchPackageJsonVersion();
-  await target.package();
-  await revertToOriginalPackageJson(backupPath);
+  const backupPath = await patchPackageJsonUpdateVersion();
+
+  try {
+    await target.package();
+  } finally {
```
Comment (nulltoken) [inline make.js:27]: @philsturgeon This should reduce even further the chance of ending up with a temporarily patched package.json in the workdir when something bad happens.

File: package.json [hunk 3]
```diff
@@ -140,10 +141,19 @@
...
     "watch": "tsc -b -w"
   },
-  "version": "0.2.2"
+  "version": "0.2.2",
+  "workspaces": {
+    "packages": [
+      "client",
+      "server"
+    ],
+    "nohoist": [
```
Comment (P0lip) [inline package.json:154]: :P OK for now, we can sync on this once you are back from holidays.

File: .github/dependabot.yml (hunk 1 above)
Comment (P0lip) [reply .github/dependabot.yml:5]: > We leverage "nohoist" (actually killing most of the workspaces benefits) so that webpack is happy and can help us reduce the size of the final vsix package by finding the modules the server don't really need

hmmm, we don't need to use `nohoist` in platform, but it's hard to tell how efficient the bundle is since we have plenty of code there.
We use some additional configuration and plugins to make sure we exclude any code that's irrelevant

> Most of the work in that change is to bridge the gap between vsce expectations about what a proper VSCode extension tree structure should look like and the impact of switching to workspaces
